spec/lib/authorizer_spec.rb

```diff
@@ -0,0 +1,189 @@
+      ]
+      expect(authorized(pals.healey_teacher) { students }).to eq [pals.healey_kindergarten_student]
+      expect(authorized(pals.shs_bill_nye) { students }).to eq [pals.shs_freshman_mari]
+    end
+
+    it 'does not work when fields are missing' do
+      test_fields = Authorizer.student_fields_for_authorization - [:id]
+
+      # Test each of the required fields individually
+      test_fields.each do |missing_field|
+        some_fields = test_fields - [missing_field]
+        students = Student.select(*some_fields).all
+
+        # Different educator permissions check different field.
+        # When this particular required field is missing, the check for
+        # one of these educators should raise an error about a missing
+        # attribute.
+        expect do
+          authorized(pals.uri) { students }
+          authorized(pals.healey_teacher) { students }
+          authorized(pals.healey_ell_teacher) { students }
+          authorized(pals.healey_sped_teacher) { students }
+          authorized(pals.shs_bill_nye) { students }
+          authorized(pals.shs_ninth_grade_counselor) { students }
+        end.to raise_error(ActiveModel::MissingAttributeError)
+      end
+    end
+  end
+
+  describe '#authorized' do
+    describe 'Student' do
+      it 'limits access with Student.all' do
+        expect(authorized(pals.uri) { Student.all }).to eq [
+          pals.healey_kindergarten_student,
+          pals.shs_freshman_mari
+        ]
+        expect(authorized(pals.healey_teacher) { Student.all }).to eq [
+          pals.healey_kindergarten_student
+        ]
+        expect(authorized(pals.shs_jodi) { Student.all }).to eq [
+          pals.shs_freshman_mari
+        ]
+        expect(authorized(pals.shs_bill_nye) { Student.all }).to eq [
+          pals.shs_freshman_mari
+        ]
+      end
+
+      it 'limits access for Student.find' do
+        expect(authorized(pals.healey_teacher) { Student.find(pals.shs_freshman_mari.id) }).to eq nil
+      end
+
+      it 'limits access for arrays' do
+        expect(authorized(pals.healey_teacher) { [pals.shs_freshman_mari, pals.healey_kindergarten_student] }).to eq [pals.healey_kindergarten_student]
+      end
+
+      it 'raises if given a Student model with `#select` filtering out fields needed for authorization' do
+        thin_student = Student.select(:id, :local_id).find(pals.shs_freshman_mari.id)
+        expect(authorized(pals.uri) { thin_student }.attributes).to eq({
+          'id' => pals.shs_freshman_mari.id,
+          'local_id' => pals.shs_freshman_mari.local_id
+        })
+        expect { authorized(pals.healey_teacher) { thin_student } }.to raise_error(ActiveModel::MissingAttributeError)
+        expect { authorized(pals.shs_bill_nye) { thin_student } }.to raise_error(ActiveModel::MissingAttributeError)
+      end
+
+      it 'if `select` was used on an ActiveRecord::Relation for students, fields needed for authorization are added in' do
```

Comment:
The test code here is identical to the test code above. But the context block description is different. So we should either change the test code to test a different thing or remove the duplication. I can take a crack at updating the test code based on what you described @kevinrobinson.

Reply:
Good catch!  I think based on the text description, intention was to change line 102 to something like `Student.select(:id, :local_id).where(id: pals.shs_freshman_mair.id)` or something that returns an `ActiveRecord::Relation`.

Reply:
Does what I sketched out in #1303 test what you were thinking?